app/[locale]/layout.tsx

```diff
@@ -1,20 +1,78 @@
-import { useLocale } from 'next-intl';
-import { notFound } from 'next/navigation';
-import { LocalePageProps } from '../layout';
+import 'styles/globals.scss';
+import classNames from 'classnames';
+import { Inter } from 'next/font/google';
+import { PropsWithChildren } from 'react';
+import { fetchGraphql, fetchGraphqlNavigation } from 'utils/graphql';
+import { SingleTypeGeneralQuery } from 'graphql/types';
+import { SINGLE_TYPE_GENERAL } from 'graphql/singleTypes/general';
+import { GeneralSetting } from '../graphql/types';
+import { NavigationItem } from '@ef2/content-components-react';
+import { NextIntlClientProvider } from 'next-intl';
+import ClientLayout from '../ClientLayout';
+import Header from 'components/layout/Header';
+import Footer from 'components/layout/Footer';
+import {notFound} from 'next/navigation';
+ 
+export function generateStaticParams() {
+  return [{locale: 'en'}, {locale: 'de'}];
+}
 
-const Layout = ({ children, params }: LocalePageProps) => {
-    const locale = useLocale();
+const inter = Inter({
+    // Only the design weights.
+    weight: ['300', '400', '500', '600', '700'],
+    variable: '--inter',
+    display: 'swap',
+    subsets: ['latin']
+});
 
-    // Show a 404 error if the user requests an unknown locale
-    if (params.locale !== locale) {
-        notFound();
+interface LayoutDataProps {
+    generalSettings: GeneralSetting;
+    navigation: NavigationItem[];
+}
+
+export const getLayoutData = async (locale?: string): Promise<LayoutDataProps> => {
+    const [settings, navigation] = await Promise.all([
+        fetchGraphql<SingleTypeGeneralQuery>(SINGLE_TYPE_GENERAL, { locale }),
+        fetchGraphqlNavigation('main-navigation', locale ?? process.env.NEXT_PUBLIC_DEFAULT_LOCALE!)
+    ]);
+
+    const generalSettings = settings.generalSetting?.data?.attributes!;
+
+    return { generalSettings, navigation };
+};
+
+const getMessages = async (locale: string) => {
```

Comment:
deze is niet meer nodig, snap de functie maar met de catch vind ik zelf iets mooier